internal/backend/runtime/omni/controllers/omni/cluster_machine_config_status.go

```diff
@@ -465,7 +465,12 @@ func (h *clusterMachineConfigStatusControllerHandler) syncInstallImageAndSchemat
 		return false, err
 	}
 
-	_, err = c.Upgrade(upgradeCtx, image, !maintenance, stageUpgrade, false)
+	_, err = c.UpgradeWithOptions(upgradeCtx,
```

Comment:
this is the most tricky change.

@frezbo please don't change machinery's client interfaces - that is going to break consumers -- even if Talos itself doesn't use some option, it doesn't mean that there are no other consumers

Reply:
should I revert?

Reply:
yes, let's do that please for 1.10.1

Reply:
this Omni change should be fine though - using better API